### PR TITLE
NO-ISSUE: Allow agent-vm and image targets

### PR DESCRIPTION
**Pull request by mangelajo**
This commit enables the agent-vm and image building, it requires sudo root privileges, and libvirtd/virt-install tools.

### CODE REVIEW

File: deploy/agent-vm.mk [hunk 1]
```diff
@@ -0,0 +1,28 @@
+bin/output/disk.qcow2: hack/Containerfile.local
```
Comment (mangelajo) [inline deploy/agent-vm.mk:1]:
```suggestion
bin/output/disk.qcow2: hack/Containerfile.local rpm
```

File: deploy/agent-vm.mk [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+agent-image: bin/output/disk.qcow2
+	@echo "Agent image built at bin/output/disk.qcow2"
+
+.PHONY: agent-image
+
+agent-vm: bin/output/qcow2/disk.qcow2
+	@echo "Booting Agent VM from bin/output/qcow2/disk.qcow2"
+	sudo cp bin/output/qcow2/disk.qcow2 /var/lib/libvirt/images/disk.qcow2
+	sudo chown qemu:qemu /var/lib/libvirt/images/disk.qcow2
+	sudo virt-install --name flightctl-device --vcpus 2 --memory 1024     --import --disk /var/lib/libvirt/images/disk.qcow2,format=qcow2 \
```
Comment (hexfusion) [inline deploy/agent-vm.mk:18]: minor: can we put each flag on a newline  please :)

Comment (hexfusion) [reply deploy/agent-vm.mk:18]: minor: Makefile take a param for VIRT_DISK or similar?

**Review by hexfusion — APPROVED** [review]
awesome :heart:

File: deploy/agent-vm.mk (hunk 2 above)
Comment (mangelajo) [reply deploy/agent-vm.mk:18]: hmmm we can accept a VMNAME to let start multiple instances

File: deploy/agent-vm.mk [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+					quay.io/centos-bootc/bootc-image-builder:latest \
+					--type qcow2 \
+					--local localhost/local-flightctl-agent:latest
+
+agent-image: bin/output/disk.qcow2
+	@echo "Agent image built at bin/output/disk.qcow2"
+
+.PHONY: agent-image
+
+agent-vm: bin/output/qcow2/disk.qcow2
```
Comment (oglok) [inline deploy/agent-vm.mk:26]: `make agent-vm` is the entry target for this, right? If so, it'll check if that file exists (or go to a target that generates it), but there isn't. right?

File: deploy/agent-vm.mk [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+					--privileged \
+					--pull=newer \
+					--security-opt label=type:unconfined_t \
+	                -v $(shell pwd)/bin/output:/output \
+					-v /var/lib/containers/storage:/var/lib/containers/storage \
+					quay.io/centos-bootc/bootc-image-builder:latest \
+					--type qcow2 \
+					--local localhost/local-flightctl-agent:latest
+
+agent-image: bin/output/disk.qcow2
```
Comment (oglok) [inline deploy/agent-vm.mk:21]:
```suggestion
agent-image: bin/output/qcow2/disk.qcow2
```

**Pull request merged**